ipfs-pinning-service.yaml

```diff
@@ -141,66 +193,134 @@ paths:
 
 components:
   schemas:
-    Error:
+
+    PinStatus:
+      description: Pin object with Status
       type: object
       properties:
-        code:
-          type: integer
-        message:
+        id:
+          description: CID of Pin object that can be used for status checks of ongoing pinning
           type: string
+        status:
+          $ref: '#/components/schemas/Status'
+        pin:
+          $ref: '#/components/schemas/Pin'
 
-    Cid:
+    Pin:
+      description: Pin object
       type: object
       properties:
-        /:
+        cid:
+          description: CID to be pinned
           type: string
+        depth:
+          description: Defines how deep the DAG should be pinned (-1 recursive, 0 direct)
```

Comment:
Could this be recursive by default so `depth` is only used if it's specified?

We wouldn't need to deal with `depth: -1` then, which just looks a bit weird.

Reply:
Is there a reason we need to filter based on depth? Having a recursive / direct depth filter wouldn't be possible on our end right now. I don't think any pinning service that currently exists keeps track of anything other than the root CID.